homeassistant/components/sabnzbd/config_flow.py

```diff
@@ -0,0 +1,80 @@
+"""Adds config flow for SabNzbd."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PATH,
+    CONF_PORT,
+    CONF_SSL,
+)
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_HOST, DEFAULT_NAME, DEFAULT_PORT, DEFAULT_SSL, DOMAIN
+from .errors import AuthenticationError
+from .sab import get_client
+
+_LOGGER = logging.getLogger(__name__)
+
+USER_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
```

Comment:
Why request a name, can't we get name from the server ?

Reply:
sabnzbd client doesn't expose that.